### PR TITLE
Making --help flag work

### PR DESCRIPTION
**Pull request by tvillaren**
# Why

As described in https://github.com/fabien0102/ts-to-zod/issues/196, the `--help` flag didn't work anymore. 
Trying to add tests for `oclif`, I updated: 
- `@oclif/core` to latest (`3.26.0`)
- the `run` script, following the templates provided on oclif [website](https://oclif.io/docs/templates) (split into a `run` and `dev` ones)
- adding the `strategy` as `single` command for oclif ([documentation](https://oclif.io/docs/single_command_cli))

# Result

Running `ts-to-zod --help` will output the help information:

```
Generate Zod schemas from a Typescript file

USAGE
  $ ts-to-zod  --all
  $ ts-to-zod  --config example
  $ ts-to-zod  --config example/person
  $ ts-to-zod  --config config

ARGUMENTS
  INPUT   input file (typescript)
  OUTPUT  output file (zod schemas)

FLAGS
  -a, --all                    Execute all configs
  -c, --config=<option>        Execute one config
                               <options: example|example/person|config>
  -h, --help                   Show CLI help.
  -i, --init                   Create a ts-to-zod.config.js file
  -k, --keepComments           Keep parameters comments
  -v, --version                Show CLI version.
  -w, --watch                  Watch input file(s) for changes and re-run related task
      --inferredTypes=<value>  Path of z.infer<> types file
      --skipParseJSDoc         Skip the creation of zod validators from JSDoc annotations
      --skipValidation         Skip the validation step (not recommended)

DESCRIPTION
  Generate Zod schemas from a Typescript file

EXAMPLES
  $ ts-to-zod src/types.ts src/types.zod.ts
```

# Tests

Tested locally on the project I use `ts-to-zod` and it does seem to work. 
Still trying to get `@oclif/test` to work but I believe we can merge this still

### CODE REVIEW

File: bin/dev
Comment (schiller-manuel) [inline bin/dev:1]: how is this executed? 
package.json `bin` points to:

`"ts-to-zod": "bin/run"`

Comment (tvillaren) [reply bin/dev:1]: This if for development purpose only, you can run it when developing on your machine via `./bin/dev --all` for instance. 
The `./bin/run` command needs the lib to be built to be able to run. 

IMO, splitting both makes it clearer and forces to have a clean release pipeline.

Comment (tvillaren) [reply bin/dev:1]: I just added ([bdab889](https://github.com/fabien0102/ts-to-zod/pull/222/commits/bdab8899635aaaae4142b7288f9c660b75731ed6)) a `yarn build` step to the test pipeline: it seems the CI tests were running on the uncompiled version of the lib and thus failed after my split.

**Review by schiller-manuel — APPROVED** [review]
LGTM

**Pull request merged**